### PR TITLE
Make Labels and Annotation consistent

### PR DESCRIPTION
**Pull request by the-simmon**
This PR fixes https://github.com/secureCodeBox/secureCodeBox/issues/1200

### CODE REVIEW

File: operator/controllers/execution/scans/parse_reconciler.go [hunk 1]
```diff
@@ -103,9 +103,13 @@ func (r *ScanReconciler) startParser(scan *executionv1.Scan) error {
 			BackoffLimit:            &backOffLimit,
 			Template: corev1.PodTemplateSpec{
 				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app.kubernetes.io/managed-by=securecodebox": "securecodebox",
+					},
```
Comment (malexmave) [inline operator/controllers/execution/scans/parse_reconciler.go:108]: Is this pattern correct? `managed-by=securecodebox: securecodebox` seems redundant.

**Comment by malexmave** [comment]
Thanks for your work! One request: Would it also be possible to add `securecodebox.io/job-type: scanner | parser | hook | operator | ...` labels in all relevant places? Currently, this seems to be handled as an annotation, and not for everything in SCB (see #1200). Making this a label would allow us to filter pods based on it using a podSelector, which I don't think is possible for annotations. A use case for this filtering would be in clusters using least-privilege networking with [Network Policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/) - here, it may be desireable to permit egress to the Internet only for hooks, but not scanners and parsers (for example).

Although I am not sure if removing the annotation and turning it into a label would be considered a breaking change. Probably yes?

Unrelated: Do we have any idea why all the CI runs are failing (timing out) here? In some cases, it looks like they are generating good results, but the results are not pulled in correctly?

**Comment by J12934** [comment]
> Although I am not sure if removing the annotation and turning it into a label would be considered a breaking change. Probably yes?

@malexmave 

Doesn't seem to be breaking (anymore not sure about before?) as this only adds new labels and doesn't remove anything.

@the-simmon are all remarks from Max addressed, code wise this looks good to me 👍

**Comment by J12934** [comment]
@the-simmon very nice, thank you 👍

can you create a ticket for the annotations to remove and add them for the v4 board?

**Pull request merged**